engines/suppliers/app/presenters/suppliers/producer_presenter.rb

```diff
@@ -0,0 +1,44 @@
+module Suppliers
+  class ProducerPresenter
+    # @param producer [Producer]
+    # @param user [User]
+    def initialize(producer, user)
+      @producer = producer
+      @user = user
+    end
+
+    # Checks if the given user has edit rights on the producer
+    #
+    # @return [Boolean]
+    def can_edit
+      producer.has_admin?(user) || group_admin?
+    end
+
+    delegate :id, :name, :email, :address, :created_at, :updated_at, to: :producer
+
+    private
+
+    attr_reader :producer, :user
+
+    # Checks if the producer is associated to a group
+    # and if the user is an admin of that group
+    #
+    # @return [Boolean]
+    def group_admin?
+      return false unless group
+
+      group.has_admin?(user)
```

Comment:
`group && group.has_admin?(user)` right? and yet another N+1, do the presenters eager load some how?